ambry-tools/src/main/java/com.github.ambry/account/AccountTool.java

```diff
@@ -0,0 +1,542 @@
+   * @param zkServer The address to zookeeper server.
+   * @param storePath The root path for {@link HelixPropertyStore}.
+   * @param zkConnectionTimeoutMs The connection timeout for {@link HelixPropertyStore}.
+   * @param zkSessionTimeoutMs The session timeout for {@link HelixPropertyStore}.
+   * @return {@link VerifiableProperties}.
+   */
+  private VerifiableProperties getVerifiableProperties(String zkServer, String storePath, int zkConnectionTimeoutMs,
+      int zkSessionTimeoutMs) {
+    Properties properties = new Properties();
+    // for creating HelixAccountService
+    properties.setProperty(HelixPropertyStoreConfig.HELIX_ZK_CLIENT_CONNECTION_TIMEOUT_MS,
+        String.valueOf(zkConnectionTimeoutMs));
+    properties.setProperty(HelixPropertyStoreConfig.HELIX_ZK_CLIENT_SESSION_TIMEOUT_MS,
+        String.valueOf(zkSessionTimeoutMs));
+    properties.setProperty(HelixPropertyStoreConfig.HELIX_ROOT_PATH, storePath);
+    properties.setProperty(HelixAccountServiceConfig.ZK_CLIENT_CONNECT_STRING_KEY, zkServer);
+
+    // for creating NonBlockingRouter
+    properties.setProperty("clustermap.dcs.zk.connect.strings", clusterMapDcsZkConnectString);
+    properties.setProperty("clustermap.clusteragents.factory", "com.github.ambry.clustermap.HelixClusterAgentsFactory");
+    properties.setProperty("clustermap.cluster.name", clusterName);
+    properties.setProperty("clustermap.host.name", hostname);
+    properties.setProperty("clustermap.datacenter.name", dcName);
+    properties.setProperty("router.hostname", hostname);
+    properties.setProperty("router.datacenter.name", dcName);
+    properties.setProperty("kms.default.container.key",
+        "B375A26A71490437AA024E4FADD5B497FDFF1A8EA6FF12F6FB65AF2720B59CCF");
+    return new VerifiableProperties(properties);
+  }
+
+  /**
+   * Create a {@link RouterStore}. This router store will overwrite the account metadata whene updating
+   * the account metadata.
+   * @return {@link RouterStore}.
+   * @throws Exception Any unexpected exception.
+   */
+  private RouterStore getRouterStore() throws Exception {
+    return new RouterStore(new AccountServiceMetrics(registry), backupFileManager, helixStore,
+        new AtomicReference<>(getRouter()), true, 100);
+  }
+
+  /**
+   * Create a {@link Router} for {@link RouterStore}.
+   * @return The {@link Router}.
+   * @throws Exception Any unexpected exception.
+   */
+  private Router getRouter() throws Exception {
+    // Create a HelixAccountService for the router.
+    AccountServiceFactory accountServiceFactory =
+        Utils.getObj("com.github.ambry.account.HelixAccountServiceFactory", verifiableProperties, registry);
+    accountService = (HelixAccountService) accountServiceFactory.getAccountService();
+
+    ClusterMapConfig clusterMapConfig = new ClusterMapConfig(verifiableProperties);
+    ClusterAgentsFactory clusterAgentsFactory =
+        Utils.getObj(clusterMapConfig.clusterMapClusterAgentsFactory, clusterMapConfig, "", "");
```

Comment:
I suggest passing PartitionLayout and HardwareLayout as well. I know it is not needed for HelixClusterManager but it's better to support update account when we are using StaticClusterManager.

Reply:
Added hardware layout and partition layout as optional parameter.